### DISCUSSION TITLE
webpack node api don't run the same as command line webpack

### DISCUSSION THREAD

**Discussion by lv2lrn4lf** [webpack/webpack#15599] [General]
Simple, the nodejs webpack api doesn't run the same way as the webpack command line.

I'm trying to write custom loaders and plugins, I'm stuck at the loaders and after banging my head, i see that it doesn't even work the same way as the command line one does. Why is this? How can i make the webpack node API behavior is the same way as the webpack command line?

I want to get the file content from the loader and it doesn't pass that when call via node API, it pass a object 

`{resouce:'/path/to/file.js',...//other stuff}`

instead of a file i get the above. How do i get the file while running webpack with the nodejs API?

**Comment by vankop** [comment]
Thank you for creating this issue. However, issues need to follow one of our templates so that we can clearly understand your particular circumstances.

Please help us by recreating the issue using one of our templates.

Please fill out all fields, we can't help you without it. Also will be very useful creating reproducible repo.

**Comment by lv2lrn4lf** [comment]
I request for this to be deleted or move to the discussion which I didn't know about.

I want this move to discussion and will make a new post if it is not moved there and notify over there that I made this and see will anyone delete it since it'll be a duplicate.
